### PR TITLE
feat(rust, python): repeat_by should also support broadcasting of LHS

### LINKED ISSUE #10702: Fix broadcasting for `repeat_by`
### Problem description

The result of
```
pl.int_ranges(0, 3)
```
can be achieved by 
```
pl.int_range(0, 3).implode()
```

There is also pl.repeat, which doesn't have any pl.repeat**s**

Maybe I am missing something (performance, ...), but it makes no sense in this case.

### PR DESCRIPTION
**Pull request by reswqa**
This fixes #10702.

### CODE REVIEW

File: crates/polars-core/src/chunked_array/ops/repeat_by.rs [hunk 1]
```diff
@@ -22,95 +22,127 @@ where
...
+        match (self.len(), by.len()) {
+            (left_len, right_len) if left_len == right_len => {
+                Ok(arity::binary(self, by, |arr, by| {
+                    let iter = arr.into_iter().zip(by).map(|(opt_v, opt_by)| {
+                        opt_by.map(|by| std::iter::repeat(opt_v.copied()).take(*by as usize))
+                    });
+
+                    // SAFETY: length of iter is trusted.
+                    unsafe {
+                        LargeListArray::from_iter_primitive_trusted_len(
```
Comment (reswqa) [inline crates/polars-core/src/chunked_array/ops/repeat_by.rs:34]: I believe we can share most of the code with these types, but before that, maybe we should unify this call first. 🤔

**Comment by cmdlineluser** [comment]
Thank you @reswqa!

I just remembered there was a previous issue opened for this which can also be closed.

#9815

**Comment by reswqa** [comment]
Not sure if this is by design or we can improve it. But we mark it as accepted, so open this pr first ：)

File: crates/polars-core/src/chunked_array/ops/repeat_by.rs (hunk 1 above)
Comment (ritchie46) [reply crates/polars-core/src/chunked_array/ops/repeat_by.rs:34]: Yes, might be a good follow up!

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**